### PR TITLE
Add LLVM based MinGW targets

### PR DESCRIPTION
**Pull request by mati865**
Rust PR: https://github.com/rust-lang/rust/pull/94872

### CODE REVIEW

**Comment by dot-asm** [comment]
This is duplicate of [comment at rust-lang](https://github.com/rust-lang/rust/pull/94872#issuecomment-1083683807)

Not that my word weighs a lot here and I apologize, but I fail to see the need for an additional target. Is there a programming interface that `clang` uses/requires that `gcc` and/or Microsoft `cl` don't? What I'm trying to say is that it's rather the programming interface that defines any particular target platform, not the compiler. Compilers are supposed to be and actually are interchangeable. For example, on Windows you can use `clang-cl` instead of vendor `cl` with `*-msvc` targets. On Linux you can freely switch between `clang` and `gcc`. And I see no reason for why it would be impossible to have the similar arrangement for `*-windows-gnu` targets.

**Comment by dot-asm** [comment]
Just in case for reference, my remark was refuted in the referred thread at rust-lang. To summarize, it boils down to irreconcilable differences in exception handling mechanisms.

File: src/lib.rs [hunk 1]
```diff
@@ -2546,7 +2550,7 @@ impl Build {
             .as_ref()
             .map(|s| s.trim_right_matches('-').to_owned());
         cross_compile.or(match &target[..] {
-            "aarch64-pc-windows-gnu" => Some("aarch64-w64-mingw32"),
+            "aarch64-pc-windows-gnullvm" => Some("aarch64-w64-mingw32"),
```
Comment (thomcc) [inline src/lib.rs:2553]: Does this mean to remove the `aarch64-pc-windows-gnu` target?

**Review by thomcc — CHANGES_REQUESTED** [review]
I'm in favor of supporting the `-gnullvm` targets, but this seems to remove one of the `-gnu` targets?

File: src/lib.rs (hunk 1 above)
Comment (mati865) [reply src/lib.rs:2553]: Yes, there is no AArch64 support for Windows in both Binutils and GCC. If you insist I can restore it but it's dead code.

Comment (thomcc) [reply src/lib.rs:2553]: Can you add a comment to this effect, then?

Comment (mati865) [reply src/lib.rs:2553]: Comment in commit message/PR description or something else?
There was no `aarch64-pc-windows-gnu` target ever available in Rust.

Comment (thomcc) [reply src/lib.rs:2553]: Before that arm, something like this maybe?
```rs
// Note: there is no `aarch64-pc-windows-gnu` target, only `-gnullvm`
```

**Comment by mati865** [comment]
Possibly due to repo move GitHub ignores pushes to this branch, moving to new PR: https://github.com/rust-lang/cc-rs/pull/734

**Pull request closed without merging**